### ISSUE TITLE
Writer does not support field names with special characters

### ISSUE THREAD

**Issue by blkerby** [noirello/pyorc#25]
I'm seeing that field names containing special characters lead to an error when constructing a Writer. Here is an example, with a field name "abc.def" containing a period:

```
import pyorc
import io

schema = pyorc.Struct(**{"abc.def": pyorc.Int()})
writer = pyorc.Writer(io.BytesIO(), schema)
```

It leads to this error:

```
RuntimeError: Unrecognized character.
```
It appears that what is happening is that the `TypeDescription` in Python is getting [converted into a string](https://github.com/noirello/pyorc/blob/13cdf3e2fc3347ea13a40a149e419bf2228dfffd/src/_pyorc/Writer.cpp#L19) (here `struct<abc.def:int>`) and then parsed on the C++ side, but in this process there is no way to quote/escape the field names so that special characters could pass through. 

I am thinking we could solve this by instead creating the C++ `Type` directly from the Python `TypeDescription` by walking over it recursively, using the C++ `orc` functions/methods like `createPrimitiveType`, `createStructType`, `addChildType`, `addStructField`, etc. Basically this would be an inverse of [`createTypeDescription`](https://github.com/noirello/pyorc/blob/13cdf3e2fc3347ea13a40a149e419bf2228dfffd/src/_pyorc/Reader.cpp#L9) in the `pyorc` reader. This would require more code than the existing solution but may be more robust than using a string as an intermediate representation here. I'd be happy to work on this and create a pull request for it, if this approach seems good?

**Comment by noirello** [comment]
Is column names with dot are allowed in ORC files?

The C++ Core seems to only expecting [alphanumeric](https://github.com/apache/orc/blob/master/c%2B%2B/src/TypeImpl.cc#L646) characters in column names.
On the other hand, the Java counterpart does have some [backticking mechanism](https://github.com/apache/orc/blob/f4eeb65a6b6c6d5476bd3a9a2f7c812a4a833177/java/core/src/java/org/apache/orc/impl/ParserUtils.java#L84), when it comes to parsing names.

I certainly didn't expected [in this method](https://github.com/noirello/pyorc/blob/master/src/pyorc/typedescription.py#L247-L251).

**Comment by blkerby** [comment]
As far as I can tell, the [ORC specification](https://orc.apache.org/specification/ORCv1/) does not put any restrictions on the characters in field names, so it would seem that an arbitrary protobuf "string" is allowed, which would be any UTF-8 with a length restriction of 2^32 bytes. By creating a `Type` programmatically using methods like `addStructField`, the C++ Core allows arbitrary characters in field names and can read and write them without issue. The only thing is that its parser function `buildTypeFromString` does not have an escaping mechanism and so can only deal with alphanumeric (plus underscore) characters, but this seems like a restriction of the C++ parser only and not a restriction on the ORC files themselves. Probably it would make sense if `buildTypeFromString` could be extended with a quoting/escaping mechanism using backticks like in the Java implementation.

**Comment by noirello** [comment]
Looking the source and the docs, I've come to the same conclusion.

**Comment by blkerby** [comment]
Fixed by #26.
